packages/core/src/lib/markdown-it/plugins/markdown-it-icons.ts

```diff
@@ -1,30 +1,45 @@
-const octicons = require('@primer/octicons');
+import octicons from '@primer/octicons';
+import { IconName } from '@primer/octicons';
+import cheerio from "cheerio";
 
-module.exports = require('markdown-it-regexp')(
+function getOcticonIcon(iconName: string) {
+    const key: IconName | undefined = (Object.keys(octicons) as Array<IconName>).find(key => key === iconName);
+    if (key === undefined) {
+        return null;
+    }
+    return octicons[key];
+}
+
+export = require('markdown-it-regexp')(
   /:(fa[brs]|glyphicon|octicon|octiconlight|mi[forst])-([a-z-]+)~?([a-z-]+)?:/,
-  (match) => {
+  (match: string[]) => {
     const iconFontType = match[1];
     const iconFontName = match[2];
     const iconClass = match[3];
 
     if (iconFontType === 'glyphicon') {
       return `<span aria-hidden="true" class="glyphicon glyphicon-${iconFontName}"></span>`;
     } else if (iconFontType === 'octicon') {
+      const octiconIcon = getOcticonIcon(iconFontName);
       // ensure octicons are valid
-      if (!(iconFontName in octicons)) {
-        return '<span aria-hidden="true"></span>';
+      if (octiconIcon === null) {
+          return '<span aria-hidden="true"></span>';
       }
       return iconClass
-        ? octicons[iconFontName].toSVG({ class: iconClass })
-        : octicons[iconFontName].toSVG();
+        ? octiconIcon.toSVG({ class: iconClass })
+        : octiconIcon.toSVG();
     } else if (iconFontType === 'octiconlight') {
-      // ensure octicons are valid
-      if (!(iconFontName in octicons)) {
-        return '<span aria-hidden="true"></span>';
-      }
-      return iconClass
-        ? octicons[iconFontName].toSVG({ style: 'color: #fff;', class: iconClass })
-        : octicons[iconFontName].toSVG({ style: 'color: #fff;' });
+        const octiconIcon = getOcticonIcon(iconFontName);
+        // ensure octicons are valid
+        if (octiconIcon === null) {
+            return '<span aria-hidden="true"></span>';
+        }
+        const octiconIconHtml = iconClass
+            ? octiconIcon.toSVG({ class: iconClass })
+            : octiconIcon.toSVG();
+        const $ = cheerio.load(octiconIconHtml);
+        $('svg').attr('style', 'color: #fff');
+        return $.html();
```

Comment:
Is there a reason why we are performing this transformation in cheerio instead of passing `style` as an option in the `toSVG` method?

Reply:
the interface of SVG options doesn't have the attribute style😭